src/pages/settings/company/edit/CompanyEdit.tsx

```diff
@@ -0,0 +1,220 @@
+
+      try {
+        const response = await request(
+          'GET',
+          endpoint('/api/v1/companies/:id', { id: companyId })
+        );
+
+        const respondedCompany = response.data.data;
+
+        const companySettings = respondedCompany.settings;
+
+        respondedCompany.settings = {
+          ...respondedCompany.settings,
+          name: name ? name : companySettings?.name,
+          subdomain: subdomain ? subdomain : companySettings?.subdomain,
+          language_id: language_id ? language_id : companySettings?.language_id,
+          currency_id: currency_id ? currency_id : companySettings?.currency_id,
+        };
+
+        if (subdomain && isHosted()) {
+          await request('POST', endpoint('/api/v1/check_subdomain'), {
+            subdomain: subdomain,
+          });
+        }
+
+        await request(
+          'PUT',
+          endpoint('/api/v1/companies/:id', { id: companyId }),
+          respondedCompany
+        );
+
+        setIsFormBusy(false);
+
+        toast.success('updated_company');
+
+        props.setIsModalOpen(false);
+
+        window.location.href = route('/');
+      } catch (error) {
+        const axiosError = error as AxiosError;
+        console.error(axiosError);
+        if (axiosError?.response?.status === 422) {
+          setErrors(axiosError?.response?.data?.errors);
+          toast.dismiss();
+        } else {
+          toast.error();
+        }
+        if (axiosError?.response?.status === 401) {
+          toast.error(t('subdomain_is_not_available'));
+        }
+        setIsFormBusy(false);
+      }
+    }
+  };
+
+  useEffect(() => {
+    if (currentCompany?.id && !companyId) {
+      setCompanyId('');
+    }
+    if (companyId === '') {
+      setCompanyId(currentCompany?.id);
+      fetchCompanyDetails(currentCompany?.id);
+    }
+  }, [currentCompany]);
+
+  return (
+    <Modal
+      title={t('company_details')}
+      visible={props.isModalOpen}
+      onClose={() => {
+        props.setIsModalOpen(false);
+        setErrors(undefined);
+      }}
+      size="regular"
+      backgroundColor="white"
+    >
+      <div>
+        <div className="flex justify-between mb-6 px-6">
+          <h1 className="text-2xl text-gray-900">
+            {t('welcome_to_invoice_ninja')}
+          </h1>
+        </div>
+        <Element leftSide={t('company_name')}>
+          <InputField
+            value={company?.name}
+            onValueChange={(value) => handleChange('name', value)}
+            errorMessage={errors?.errors?.name}
+          />
+        </Element>
+        {isHosted() && (
+          <Element leftSide={t('subdomain')}>
+            <InputField
+              value={company?.subdomain}
+              onValueChange={(value) => handleChange('subdomain', value)}
+            />
+          </Element>
+        )}
+        <Element leftSide={t('language')}>
+          <SelectField
+            defaultValue={languages[0]?.id}
```

Comment:
This is not good. I noted that the default language should be English, not Albanian or any other that is first.

Reply:
> This is not good. I noted that the default language should be English, not Albanian or any other that is first.

Regarding the logic on the API, when we create company, in the database will always be stored English as default. So, this is unnecessary. I removed it.